src/serve.ts

```diff
@@ -0,0 +1,35 @@
+import fs from 'fs';
+
+import { Context, Middleware } from '@curveball/core';
+
+import { doesMatchRoute, validateFile, getMimeType } from './util';
+
+const fsPromises = fs.promises;
+
+export type Options = {
+  staticDir: string,
+}
+
+export function serveFiles(options: Options): Middleware {
+  return async (ctx: Context, next) => {
+    await serve(options, ctx);
+    next();
+  };
+}
+
```

Comment:
Should not unconditionally call next. `next`. Also this, should be either `return next()` or `await next()`